src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerCleanerTest.java

```diff
@@ -0,0 +1,135 @@
+        StorageClientsHelper.setAzureTestMode();
+
+        setupClass();
+    }
+
+    @BeforeEach
+    void setUp() {
+        dbHelper.deleteAll();
+    }
+
+    @Override
+    protected void beforeTest() {
+        BlobServiceClient storageClient = StorageClientsHelper.getStorageClient(interceptorManager);
+        containerCleaner = new ContainerCleaner(storageClient, envelopeRepository);
+    }
+
+    @Test
+    void should_not_find_any_blobs() {
+        assertThatCode(() -> containerCleaner.process(CONTAINER_NAME)).doesNotThrowAnyException();
+    }
+
+    @Test
+    void should_find_blobs_and_delete() {
+        // given
+        String[] dispatchedFileNames = new String[]{"file1.zip", "file3.zip"};
+        createEnvelopes(DISPATCHED, dispatchedFileNames);
+
+        // when
+        assertThatCode(() -> containerCleaner.process(CONTAINER_NAME)).doesNotThrowAnyException();
+
+        // then
+        assertFilesIsDeleteState(true, dispatchedFileNames);
+    }
+
+    @Test
+    void should_delete_only_dispatched_blobs() {
+        // given
+        String[] dispatchedFileNames = new String[]{"file1.zip", "file3.zip"};
+        String[] rejectedFileNames = new String[]{"file2.zip", "file4.zip"};
+        createEnvelopes(DISPATCHED, dispatchedFileNames);
+        createEnvelopes(REJECTED, rejectedFileNames);
+
+        // when
+        assertThatCode(() -> containerCleaner.process(CONTAINER_NAME)).doesNotThrowAnyException();
+
+        // then
+        assertFilesIsDeleteState(true, dispatchedFileNames);
+        assertFilesIsDeleteState(false, rejectedFileNames);
+    }
+
+    @Test
+    void should_handle_non_exiting_file() {
+        // given
+        String[] dispatchedFileNames = new String[]{"causes_404.zip"};
+        createEnvelopes(DISPATCHED, dispatchedFileNames);
+
+        // when
+        assertThatCode(() -> containerCleaner.process(CONTAINER_NAME)).doesNotThrowAnyException();
+
+        // then
+        assertFilesIsDeleteState(true, dispatchedFileNames);
+    }
+
+    @Test
+    void should_handle_server_error() {
+        // given
+        String[] dispatchedFileNames = new String[]{"causes_500.zip"};
+        createEnvelopes(DISPATCHED, dispatchedFileNames);
+
+        // when
+        assertThatCode(() -> containerCleaner.process(CONTAINER_NAME)).doesNotThrowAnyException();
+    }
+
+    private void createEnvelopes(Status status, String... fileNames) {
+        for (String fileName : fileNames) {
+            NewEnvelope envelope = new NewEnvelope(
+                CONTAINER_NAME,
+                fileName,
+                Instant.now(),
+                Instant.now(),
+                status
+            );
+            envelopeRepository.insert(envelope);
+        }
+    }
+
+    private void assertFilesIsDeleteState(boolean isDeleted, String... fileNames) {
+        for (String fileName : fileNames) {
+            assertThat(envelopeRepository.find(fileName, CONTAINER_NAME).isPresent()).isTrue();
```

Comment:
batter to run assertion on the optional object (will result in a clearer error message)

Reply:
To be exact:
```java
assertThat(envelopeRepository.find(fileName, CONTAINER_NAME)).hasValueSatisfying(env -> {
    assertThat(env.isDeleted).isEqualTo(isDeleted)
})
```